machine/v1beta1/types_azureprovider.go

```diff
@@ -183,6 +183,19 @@ type OSDisk struct {
 	ManagedDisk ManagedDiskParameters `json:"managedDisk"`
 	// DiskSizeGB is the size in GB to assign to the data disk.
 	DiskSizeGB int32 `json:"diskSizeGB"`
+	// DiffDiskSettings describe ephemeral disk settings for the os disk.
+	// +optional
+	DiffDiskSettings DiffDiskSettings `json:"diffDiskSettings,omitempty"`
```

Comment:
Why "Diff"?

Reply:
I think this is named like this for a better match with azure API https://github.com/Azure/azure-sdk-for-go/blob/main/services/compute/mgmt/2018-06-01/compute/models.go#L1079

Reply:
> I think this is named like this for a better match with azure API https://github.com/Azure/azure-sdk-for-go/blob/main/services/compute/mgmt/2018-06-01/compute/models.go#L1079

That doesn't seem strong enough on its own.  Do we have a reason why this isn't simply DiskSettings?

Reply:
They are just `DiskSettings` now.